packages/flutter/lib/src/material/spell_check_suggestions_toolbar.dart

```diff
@@ -0,0 +1,208 @@
+
+import 'colors.dart';
+import 'material.dart';
+import 'material_localizations.dart';
+import 'spell_check_suggestions_toolbar_layout_delegate.dart';
+import 'text_selection_toolbar_text_button.dart';
+
+// Minimal padding from all edges of the selection toolbar to all edges of the
+// viewport.
+const double _kToolbarScreenPadding = 8.0;
+const double _kHandleSize = 22.0;
+
+// Padding between the toolbar and the anchor.
+const double _kToolbarContentDistanceBelow = _kHandleSize - 3.0;
+
+// The default height of the MaterialSpellCheckSuggestionsToolbar, which
+// assumes there are the maximum number of spell check suggestions available, 3.
+const double _kDefaultToolbarHeight = 193.0;
+
+/// The default spell check suggestions toolbar for Android.
+///
+/// Tries to position itself below the [anchor], but if it doesn't fit, then it
+/// readjusts to fit above bottom view insets.
+class MaterialSpellCheckSuggestionsToolbar extends StatelessWidget {
+  /// Constructs a [MaterialSpellCheckSuggestionsToolbar].
+  const MaterialSpellCheckSuggestionsToolbar({
+    super.key,
+    required this.anchor,
+    required this.buttonItems,
+  }) : assert(buttonItems != null);
+
+  /// {@template flutter.material.MaterialSpellCheckSuggestionsToolbar.anchor}
+  /// The focal point below which the toolbar attempts to position itself.
+  /// {@endtemplate}
+  final Offset anchor;
+
+  /// The buttons that will be displayed in the spell check suggestions toolbar.
+  final List<ContextMenuButtonItem> buttonItems;
+
+  /// Builds the default Android Material spell check suggestions toolbar.
+  static Widget _spellCheckSuggestionsToolbarBuilder(BuildContext context, Widget child) {
+    return _MaterialSpellCheckSuggestionsToolbarContainer(
+      child: child,
+    );
+  }
+
+  /// Builds the button items for the toolbar based on the available
+  /// spell check suggestions.
+  static List<ContextMenuButtonItem> buildButtonItems(
+    BuildContext context,
+    EditableTextState editableTextState,
+    SuggestionSpan suggestionSpan,
+  ) {
+    final List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+
+    // Build suggestion buttons.
+    for (final String suggestion in suggestionSpan.suggestions) {
+      buttonItems.add(ContextMenuButtonItem(
+        onPressed: () {
+          editableTextState
+            .replaceComposingRegion(
+              SelectionChangedCause.toolbar,
+              suggestion,
+          );
+        },
+        label: suggestion,
+      ));
+    }
+
+    // Build delete button.
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    final ContextMenuButtonItem deleteButton =
+      ContextMenuButtonItem(
+        onPressed: () {
+          editableTextState.replaceComposingRegion(
+            SelectionChangedCause.toolbar,
+            '',
+          );
+        },
+        type: ContextMenuButtonType.delete,
+        label: localizations.deleteButtonTooltip.toUpperCase(),
```

Comment:
As mentioned above, I think this label should be defined in `getButtonLabel` and omitted here.